docs/adr/0002-use-local-datastore.md

```diff
@@ -0,0 +1,58 @@
+# 2: Use local datastore for publisher workflow
+
+Date: 2018-07-20
+
+## Context
+We're building a new publishing architecture and MVP workflow in private beta. We have taken the decision to build this in a new application and therefore we need to decide how we use the publishing-api.
+
+One of the goals of the migration effort from 2016 was to for all publishing applications to be clients of the publishing-api and this API would means that these applications would not need local datastores. This is written up in the [GOV.UK Platform Architecture Goals 2017-18+][goals]
+
+[goals]: https://docs.google.com/document/d/1Oft4akc6dZfhhOjosNPbFpcLUOUjz7YG7QPcVZi8hww/edit#heading=h.5uytjxbfoe58
+
+Since writing these goals we have built the specialist-publisher application using this pattern. The difficulties of building Rails application this way have led us to reconsider this approach.
+
+We are about to build another publishing-api client and we should now use the learnings we have to ensure that not using a local datastore is the best decision.
+
+### Option 1: No local datastore
+In this option we continue with the platform architecture goals.
+
+#### Pros
++ Single source of truth - all data is in the publishing api
++ Multiple apps can edit the same content
++ Single implementation of scheduling and publishing logic
++ Single abstraction layer
+
+#### Cons
+- Effectively building an Active Resource layer
+- Harder to resolve performance problems
+- Current workflow is opinionated and may not match needs of content
+- Validation can be difficult to explain to user
+- Hard to deploy
+
+### Option 2: Use a local datastore
+#### Pros
++ It’s standard Rails so quick to build and easy to understand
++ Easy to test, testing over an API is not easy and slow
++ Easy to iterate workflow, changing the API workflow is much harder
++ Only final state of content can be put into publishing-api - some formats will need interim data which isn’t needed for publishing
```

Comment:
Using the Publishing API for storing content doesn't exclude storing interim data locally. If it isn't needed for publishing, it probably isn't necessary to go in to the Publishing API at all.

Reply:
This goes back to the intent of the Architecture Goals, does it mean absolutely no datastore or only interim data.

Reply:
My unauthoritative view, is that it's mostly about having a single source of truth for content, not storing this locally in the Publishing application, but we should get consensus on the interpretation of the Architecture Goals somehow....

Reply:
The first paragraph Section 1 from the Architecture Goals is as follows

> The original goal was that the Publishing API was the “source of truth” for all content, and that individual publishing applications should effectively be stateless, relying on the Publishing API as their “database.”

Given the above, especially the point about statelessness I would say that it means that new publishing apps should not use a local datastore. Also @timblair is back from holiday on Wednesday so we can ask him then.